src/pages/Courses/Course.tsx

```diff
@@ -1,167 +1,238 @@
-      <Outlet />
-      <main>
-        <Container fluid className="px-md-4">
-          <Row className="mt-md-2 mb-md-2">
-            <Col className="text-center">
-              <h1>Manage Courses</h1>
-            </Col>
-            <hr />
-          </Row>
-          <Row>
-            <Col md={{ span: 1, offset: 11 }} style={{ paddingBottom: "10px" }}>
-              <Button variant="outline-success" onClick={() => navigate("new")}>
-                <RiHealthBookLine />
-              </Button>
-            </Col>
-            {showDeleteConfirmation.visible && (
-              <DeleteCourse
-                courseData={showDeleteConfirmation.data!}
-                onClose={onDeleteCourseHandler}
-              />
-            )}
-            {showCopyConfirmation.visible && (
-              <CopyCourse courseData={showCopyConfirmation.data!} onClose={onCopyCourseHandler} />
-            )}
-          </Row>
-          <Row>
-            <Table
-              showGlobalFilter={false}
-              data={formattedTableData}
-              columns={tableColumns}
-              columnVisibility={{
-                id: false,
-                institution: auth.user.role === ROLE.SUPER_ADMIN.valueOf(),
-              }}
-            />
-          </Row>
-        </Container>
-      </main>
-    </>
-  );
-};
-
-export default Courses;
+import { Row as TRow } from "@tanstack/react-table";
+import Table from "components/Table/Table";
+import useAPI from "hooks/useAPI";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { Button, Col, Container, Row } from "react-bootstrap";
+import { RiHealthBookLine } from "react-icons/ri";
+import { useDispatch, useSelector } from "react-redux";
+import { Outlet, useLocation, useNavigate } from "react-router-dom";
+import { alertActions } from "store/slices/alertSlice";
+import { RootState } from "../../store/store";
+import { ICourseResponse, ROLE } from "../../utils/interfaces";
+import { courseColumns as COURSE_COLUMNS } from "./CourseColumns";
+import CopyCourse from "./CourseCopy";
+import DeleteCourse from "./CourseDelete";
+import { formatDate, mergeDataAndNamesAndInstructors } from "./CourseUtil";
+import CourseAssignments from "./CourseAssignments";
+import { ICourseResponse as ICourse } from "../../utils/interfaces";
+
+/**
+ * Courses Component: Displays and manages courses, including CRUD operations.
+ */
+
+const Courses = () => {
+  const { error, isLoading, data: CourseResponse, sendRequest: fetchCourses } = useAPI();
+  const { data: InstitutionResponse, sendRequest: fetchInstitutions } = useAPI();
+  const { data: InstructorResponse, sendRequest: fetchInstructors } = useAPI();
+  const auth = useSelector(
+    (state: RootState) => state.authentication,
+    (prev, next) => prev.isAuthenticated === next.isAuthenticated
+  );
+  const navigate = useNavigate();
+  const location = useLocation();
+  const dispatch = useDispatch();
+
+  // State for course details modal

```

Comment:
What is the difference between Course and CourseDetails?  In general, the attributes should be in the Course object unless there is a good reason for them not to be.  And if there is a good reason, the name should reflect the reason.  Putting "details" in any identifier name is a red flag to me.